sys-block/gpart/patches/gpart-0.3.patchset

```diff
@@ -0,0 +1,91 @@
+From 83e2443543ac0162726c44de94fcbc35fa03eb01 Mon Sep 17 00:00:00 2001
+From: localanu <localanu@gmail.com>
+Date: Thu, 4 Jan 2018 07:01:46 +0000
+Subject: Try to add support to haiku
+
+
+diff --git a/autogen.sh b/autogen.sh
+old mode 100644
+new mode 100755
+diff --git a/src/disku.c b/src/disku.c
+index d50cebf..106f2ff 100644
+--- a/src/disku.c
++++ b/src/disku.c
+@@ -33,6 +33,9 @@
+ #include <sys/disk.h>
+ #endif
+ 
++#if defined(__HAIKU__)
++#include <Drivers.h>
++
+ #include <unistd.h>
+ 
+ static void geometry_from_num_sectors(struct disk_geom *g, uint64_t nsects)
+@@ -95,8 +98,21 @@ static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
+ 	g.d_nsecs = o / u;
+ 	g.d_c = g.d_nsecs / g.d_h / g.d_s;
+ }
++
++
++#elif defined(__HAIKU__)
++static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
++{
++	device_geometry dg;
++	ioctl(d->d_fd, B_GET_GEOMETRY, & dg);
```

Comment:
[`ioctl()`][0] can fail, please add code to handle that case

This could be a reference for you:
https://github.com/baruch/gpart/blob/23b651d1e9c6ae1454e5d3fe315d7d4e145661c0/src/disku.c#L54

[0]: http://pubs.opengroup.org/onlinepubs/009695399/functions/ioctl.html